### ISSUE TITLE
Improve ClayModal component accessibility

### ISSUE THREAD

**Issue by matuzalemsteles** [liferay/clay#4736]
Well, our ClayModal component only has the focus control that persists inside Modal, it was discovered a way for the screen reader to be reading the content from outside Modal when pressing the "up" key, that's because Modal doesn't have the `autoFocus` or we are not moving the focus to the Modal so that the screen reader can read the content as soon as it opens, for example: Open the modal read the dialog title.

The current implementation of ClayModal is also not adding attributes like `role="dialog"`, `aria-hidden="true"` and `aria-labelledby` to improve accessibility.

**Comment by matuzalemsteles** [comment]
We need to bring this to Clay v2 as well. I'm working on this now! I will reopen too.

**Comment by matuzalemsteles** [comment]
Fixed in #4744
